flaskr.py

```diff
@@ -34,50 +34,72 @@ def init_db():
 
 @app.teardown_appcontext
 def close_db(error):
-	'''Closes the database again at the end of th request.'''
-	if hasattr(g, 'sqlite_db'):
-		g.sqlite_db.close()
+    '''Closes the database again at the end of th request.'''
+    if hasattr(g, 'sqlite_db'):
+        g.sqlite_db.close()
 
 @app.route('/')
 def show_entries():
-	db=get_db()
-	cur = db.execute('select id, title, text from entries order by id desc')
-	entries = cur.fetchall()
-	return render_template('show_entries.html', entries=entries)
+    db=get_db()
+    cur = db.execute('select id, title, text from entries order by id desc')
+    entries = cur.fetchall()
+    return render_template('show_entries.html', entries=entries)
 
 @app.route('/add', methods=['POST'])
 def add_entry():
-	if not session.get('logged_in'):
-		abort(401)
-	db = get_db()
-	db.execute('insert into entries (title, text) values (?, ?)',
-		[request.form['title'], request.form['text']])
-	db.commit()
-	flash('New entry was successfuly posted')
-	return redirect(url_for('show_entries'))
+    if not session.get('logged_in'):
+        abort(401)
+    db = get_db()
+    db.execute('insert into entries (title, text) values (?, ?)',
+        [request.form['title'], request.form['text']])
+    db.commit()
+    flash('New entry was successfuly posted')
+    return redirect(url_for('show_entries'))
 
 @app.route('/login', methods=['GET','POST'])
 def login():
-	error=None
-	if request.method == 'POST':
-		if request.form['username'] != app.config['USERNAME']:
-			error = 'Invalid username'
-		elif request.form['password'] != app.config['PASSWORD']:
-			error = 'Invalid password'
-		else:
-			session['logged_in'] = True
-			flash('You were logged in')
-			return redirect(url_for('show_entries'))
-	return render_template('login.html', error=error)			
+    error=None
+    if request.method == 'POST':
+        if request.form['username'] != app.config['USERNAME']:
+            error = 'Invalid username'
+        elif request.form['password'] != app.config['PASSWORD']:
+            error = 'Invalid password'
+        else:
+            session['logged_in'] = True
+            flash('You were logged in')
+            return redirect(url_for('show_entries'))
+    return render_template('login.html', error=error)           
 
 @app.route('/logout')
 def logout():
-	session.pop('logged_in', None)
-	flash('You were logged out')
-	return redirect(url_for('show_entries'))
\ No newline at end of file
+    session.pop('logged_in', None)
+    flash('You were logged out')
+    return redirect(url_for('show_entries'))
+
+@app.route('/edit=<int:entry_id>', methods=['GET'])
+def show_entry(entry_id):
+    """Show a singly entry."""
+    db=get_db()
+    cur = db.execute(
+        'select title, text from entries where id = (?)', 
+        (entry_id,)
+    )
+    entries = cur.fetchall()
+    return render_template('edit_entry.html',id=entry_id,entries=entries)
+
+@app.route('/edit=<int:entry_id>', methods=['POST'])
```

Comment:
Same as above - `POST /entry=id` already implies that we try to change this record (and it will be consistent with GET method).